packages/xml-views-validation/src/validators/elements/use-of-deprecated-class.ts

```diff
@@ -0,0 +1,54 @@
+import { XMLElement } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { xmlToFQN } from "@ui5-language-assistant/logic-utils";
+import { UseOfDeprecatedClassIssue } from "../../../api";
+
+export function validateUseOfDeprecatedClass(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): UseOfDeprecatedClassIssue[] {
+  const elementTagFqn = xmlToFQN(xmlElement);
+  const ui5Class = model.classes[elementTagFqn];
+  if (ui5Class === undefined) {
+    return [];
+  }
+
+  if (ui5Class.deprecatedInfo !== undefined) {
+    const issues: UseOfDeprecatedClassIssue[] = [];
+    const commonIssueFields: UseOfDeprecatedClassIssue = {
+      kind: "UseOfDeprecatedClass",
+      message: `Deprecated UI5 Class: ${elementTagFqn} used.`,
+      severity: "warn",
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      range: undefined as any
+    };
+
+    /* istanbul ignore else - defensive programing for (currently) none reproducible scenario */
+    //   `syntax.openName` will always exist, otherwise we could not resolve
+    //   `xmlToFQN(xmlElement)` because if syntax.openName does not exist
+    //   the xmlElement.name property would not exist either.
+    if (xmlElement.syntax.openName !== undefined) {
+      issues.push({
+        ...commonIssueFields,
+        range: {
+          start: xmlElement.syntax.openName.startOffset,
+          end: xmlElement.syntax.openName.endOffset
+        }
+      });
+    }
+
+    if (xmlElement.syntax.closeName !== undefined) {
```

Comment:
Not sure we need the issue in the closing tag too... also the check in the beginning is for the name in the opening tag, we don't actually check there what the closing tag is (in case it's mismatched). And if it's the same the user will already get the issue for the opening tag.

Reply:
- Jetbrains does line-through on deprecated XML Tags even on the closing tag.
  which is a kind of validation (I don't have a problem view example to check)
- The open and close tags may be far from each other visually so there is a benefit to showing both, especially after we implement sync of the tags.
- It may be best to only show one error in the problem view, but I am not sure if the LSP apis support an error with multiple locations I will check

Reply:
Actually I think they do and this might be the best solution (it will also be required for other validations - like unique IDs). It's called `relatedInformation` on the `Diagnostic` interface (the equivalent of `CompletionItem`. By the way, this also has a tags property where you can mark the diagnostic locations as deprecated or unnecessary/unreachable, and that seems to be supported in Theia.

Reply:
Currently I am changing "range" to "offsetRange**s**" to model multiple problems
on the same issue, this should suffice to translate it to LSP "relatedInformation" in the Language Server.

In the future we may(???) want slightly different information for each "duplicated" issue, if so we will expend the API at the point.